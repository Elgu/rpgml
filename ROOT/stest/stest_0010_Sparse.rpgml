

double[] d1 = [ 0 to 7 ]; 
Output a1 = d1;

print( sparse( a1, 0, 1 ) ); print( "\n" );
print( sparse( a1, 0, 2 ) ); print( "\n" );
print( sparse( a1, 0, 2, 1 ) ); print( "\n" );
print( sparse( a1, 0, 3, 0 ) ); print( "\n" );
print( sparse( a1, 0, 3, 1 ) ); print( "\n" );
print( sparse( a1, 0, 3, 2 ) ); print( "\n" );
print( sparse( a1, 0, 3 ) ); print( "\n" );

print( "=======\n" );

double[,] d2 = [
  0, 1, 2, 3;
  4, 5, 6, 7;
  8, 9, 10, 11;
  12, 13, 14, 15
]; 
Output a2 = d2;

print( sparse( sparse( a2, 0, 1 ), 1, 1 ) ); print( "\n" );
print( sparse( sparse( a2, 0, 2 ), 1, 1 ) ); print( "\n" );
print( sparse( sparse( a2, 0, 1 ), 1, 2 ) ); print( "\n" );
print( sparse( sparse( a2, 0, 2 ), 1, 2 ) ); print( "\n" );
print( sparse( sparse( a2, 0, 2, 1 ), 1, 2    ) ); print( "\n" );
print( sparse( sparse( a2, 0, 2    ), 1, 2, 1 ) ); print( "\n" );

exit();

