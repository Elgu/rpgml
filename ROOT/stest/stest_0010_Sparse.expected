[ 0, 1, 2, 3, 4, 5, 6, 7 ]
[ 0, 2, 4, 6 ]
[ 1, 3, 5, 7 ]
[ 0, 3 ]
[ 1, 4 ]
[ 2, 5 ]
[ 0, 3 ]
=======
[ 0, 1, 2, 3; 4, 5, 6, 7; 8, 9, 10, 11; 12, 13, 14, 15 ]
[ 0, 2; 4, 6; 8, 10; 12, 14 ]
[ 0, 1, 2, 3; 8, 9, 10, 11 ]
[ 0, 2; 8, 10 ]
[ 1, 3; 9, 11 ]
[ 4, 6; 12, 14 ]
